### PR TITLE
Fix Firebase version to avoid compilation freeze on iOS

### PR DESCRIPTION
**Pull request by damien-blanchet**
## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation  changes
- [ ] Other... Please describe:

## PR Checklist
For bug fixes / features, please check if your PR fulfills the following requirements:

- [x] Testing has been carried out for the changes have been added
- [ ] Regression testing has been carried out for existing functionality
- [ ] Docs have been added / updated

## What is the purpose of this PR?

Fix Firebase version to avoid compilation freeze on iOS

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

## What testing has been done on the changes in the PR?

resolve freeze issue on ios compilation

## Other information

### CODE REVIEW

File: plugin.xml [hunk 1]
```diff
@@ -132,14 +132,14 @@
...
-				<pod name="Firebase/InAppMessaging" spec="6.28.0"/>
-				<pod name="FirebaseFirestore" git="https://github.com/invertase/firestore-ios-sdk-frameworks.git" tag="6.28.0"/>
-				<pod name="Firebase/Crashlytics" spec="6.28.0"/>
+				<pod name="Firebase/Core" spec="6.28.1"/>
+				<pod name="Firebase/Auth" spec="6.28.1"/>
+				<pod name="Firebase/Messaging" spec="6.28.1"/>
+				<pod name="Firebase/Performance" spec="6.28.1"/>
+				<pod name="Firebase/RemoteConfig" spec="6.28.1"/>
+				<pod name="Firebase/InAppMessaging" spec="6.28.1"/>
+				<pod name="Firebase/Firestore" spec="6.28.1"/>
```
Comment (dpa99c) [inline plugin.xml:141]: This pre-compiled version needs to be kept as it significantly reduces build times

**Comment by dpa99c** [comment]
Fixed in `10.1.2`

**Comment by damien-blanchet** [comment]
The pre-compiler firestore version freeze the ios compilation on my side.

Ionic:

   Ionic CLI                     : 6.10.1 (/usr/local/lib/node_modules/@ionic/cli)
   Ionic Framework               : @ionic/angular 5.2.3
   @angular-devkit/build-angular : 0.1000.3
   @angular-devkit/schematics    : 10.0.3
   @angular/cli                  : 10.0.3
   @ionic/angular-toolkit        : 2.2.0

Cordova:

   Cordova CLI       : 9.0.0 (cordova-lib@9.0.1)
   Cordova Platforms : android 8.1.0, electron 1.1.1, ios 5.1.1
   Cordova Plugins   : cordova-plugin-ionic-keyboard 2.2.0, cordova-plugin-ionic-webview 4.2.1, (and 8 other plugins)

Utility:

   cordova-res (update available: 0.15.1) : 0.14.0
   native-run                             : 1.0.0

System:

   ios-deploy : 1.10.0
   ios-sim    : 8.0.2
   NodeJS     : v12.16.1 (/usr/local/bin/node)
   npm        : 6.13.4
   OS         : macOS Catalina
   Xcode      : Xcode 11.6 Build version 11E708

File: plugin.xml (hunk 1 above)
Comment (damien-blanchet) [reply plugin.xml:141]: > This pre-compiled version of Firestore needs to be kept as it significantly reduces build times

The pre-compiler firestore version freeze the ios compilation on my side.

Ionic:

Ionic CLI : 6.10.1 (/usr/local/lib/node_modules/@ionic/cli)
Ionic Framework : @ionic/angular 5.2.3
@angular-devkit/build-angular : 0.1000.3
@angular-devkit/schematics : 10.0.3
@angular/cli : 10.0.3
@ionic/angular-toolkit : 2.2.0

Cordova:

Cordova CLI : 9.0.0 (cordova-lib@9.0.1)
Cordova Platforms : android 8.1.0, electron 1.1.1, ios 5.1.1
Cordova Plugins : cordova-plugin-ionic-keyboard 2.2.0, cordova-plugin-ionic-webview 4.2.1, (and 8 other plugins)

Utility:

cordova-res (update available: 0.15.1) : 0.14.0
native-run : 1.0.0

System:

ios-deploy : 1.10.0
ios-sim : 8.0.2
NodeJS : v12.16.1 (/usr/local/bin/node)
npm : 6.13.4
OS : macOS Catalina
Xcode : Xcode 11.6 Build version 11E708

Comment (dpa99c) [reply plugin.xml:141]: The [example project](https://github.com/dpa99c/cordova-plugin-firebasex-test) builds fine for me with `cordova-plugin-firebasex@10.1.2` in Xcode 11.5 and Xcode 12.0 Beta 2.

Comment (damien-blanchet) [reply plugin.xml:141]: I follow the readme : 
```
git clone https://github.com/dpa99c/cordova-plugin-firebasex-test.git && cd cordova-plugin-firebasex-test
npm install

cordova platform add ios
cordova run ios
```

I obtain a build failed on the same step i'm freeze on my project : 
```
PhaseScriptExecution [CP]\ Copy\ Pods\ Resources /Users/damienblanchet/Library/Developer/Xcode/DerivedData/FirebaseTest-cqjrdjtdzlwcmiehwgvhhflbbajx/Build/Intermediates.noindex/FirebaseTest.build/Debug-iphonesimulator/FirebaseTest.build/Script-71CBF5A57D83BC59C2384146.sh (in target 'FirebaseTest' from project 'FirebaseTest')
    cd /Users/damienblanchet/Sources/cordova-plugin-firebasex-test/platforms/ios
    /bin/sh -c /Users/damienblanchet/Library/Developer/Xcode/DerivedData/FirebaseTest-cqjrdjtdzlwcmiehwgvhhflbbajx/Build/Intermediates.noindex/FirebaseTest.build/Debug-iphonesimulator/FirebaseTest.build/Script-71CBF5A57D83BC59C2384146.sh
/Users/damienblanchet/Sources/cordova-plugin-firebasex-test/platforms/ios/Pods/FirebaseFirestore/FirebaseFirestore/Resources/gRPCCertificates-Cpp.bundle
error: Resource "/Users/damienblanchet/Library/Developer/Xcode/DerivedData/FirebaseTest-cqjrdjtdzlwcmiehwgvhhflbbajx/Build/Products/Debug-iphonesimulator/FirebaseInAppMessaging/InAppMessagingDisplayResources.bundle" not found. Run 'pod install' to update the copy resources script.

** BUILD FAILED **
```

Comment (dpa99c) [reply plugin.xml:141]: https://github.com/dpa99c/cordova-plugin-firebasex#cordova-cli-builds

Comment (damien-blanchet) [reply plugin.xml:141]: You are right, I applied the modifications of this paragraph and I got a build success. I will investigate on my side to try to identify the issue

**Comment by Younghun-Jung** [comment]
@dpa99c 
I updated plugin version 10.1.2
However, Build does not proceed to the next stop. It stopped...

```
PhaseScriptExecution [CP]\ Prepare\ Artifacts /Users/jung-younghun/Library/Developer/Xcode/DerivedData/buildthing_IAQ_Admin_App-aptazjsnrxneemflbzesnykikyzo/Build/Intermediates.noindex/ArchiveIntermediates/buildthing\ IAQ\ Admin\ App/IntermediateBuildFilesPath/buildthing\ IAQ\ Admin\ App.build/Release-iphoneos/buildthing\ IAQ\ Admin\ App.build/Script-781CADF9E1BA668B37B6002E.sh
    cd /Users/jung-younghun/buildit/projects/build_thing/buildthing-iaq-admin-mobile/src-cordova/platforms/ios
    /bin/sh -c \"/Users/jung-younghun/Library/Developer/Xcode/DerivedData/buildthing_IAQ_Admin_App-aptazjsnrxneemflbzesnykikyzo/Build/Intermediates.noindex/ArchiveIntermediates/buildthing\ IAQ\ Admin\ App/IntermediateBuildFilesPath/buildthing\ IAQ\ Admin\ App.build/Release-iphoneos/buildthing\ IAQ\ Admin\ App.build/Script-781CADF9E1BA668B37B6002E.sh\"
```

**Comment by jh-gh** [comment]
@Younghun-Jung Yes I have the same problem when trying to build with the command line (using firebasex@10.1.2-cli).  The build just hangs.  It looks like it may be a cocoapods issue that others have reported elsewhere:

https://sugeul.github.io/2020-06-02-xcodebuild-hang-on-CP-Prepare-artifacts/
https://github.com/CocoaPods/CocoaPods/issues/9857

Unfortunately there doesn't seem to be any official fix available at this point.

I have tried editing the generated "Pods-<app name>-artifacts.sh file and changing this line:
  cat > $ARTIFACT_LIST_FILE
to 
  :> $ARTIFACT_LIST_FILE

I can then run cordova build ios and it works.

However, as this is a generated file, my change will get overwritten any time cordova recreates the project.

I may consider trying to come up with a cordova hook to do this edit automatically ... but hoping somebody else has a better idea.

**Comment by tzappia** [comment]
I upgraded to CocoaPods latest beta, 1.10.0.beta.2 with `sudo gem install cocoapods --pre` and the issue was solved. I was previously running 1.9.3 and the line with `cat > $ARTIFACT_LIST_FILE` was present and causing the hang (cat with no arguments just sits there.) @dpa99c I don't know when the pod issue was introduced, but if you were running an older version of CocoaPods that could be why you were not able to reproduce this.

**Pull request closed without merging**